mlflow/pytorch/__init__.py

```diff
@@ -571,6 +621,8 @@ def load_model(model_uri, **kwargs):
             torch.__version__,
         )
     torch_model_artifacts_path = os.path.join(local_model_path, pytorch_conf["model_data"])
+    if load_state_dict:
+        return torch_model_artifacts_path
```

Comment:
@shrinath-suresh Thanks for the PR! With the current approach, we can't reconstruct the model when we serve it with `mlflow models serve`. I'm wondering if we should log a model class (as a python script?) along with a model.

Reply:
Logging model class would work only for the smaller models. In case of complex models, there will be multiple nested python scripts. It would be better to leave it to the user for loading the  model class.

Reply:
Can we serialize a pytorch model **class**?

Reply:
Input to the [mlflow.pytorch.load_state_dict](https://github.com/chauhang/mlflow/blob/749c99f0158dd57a48432d5e8ed0ad498944d8cf/mlflow/pytorch/__init__.py#L539) is the **object** of the model class (not the model **class** name)
 
By keeping model object as input, user has more control over creating the object from the model class (For example: constructor arguments).

Reply:
Ah we need to log constructor arguments to instantiate the model properly from state_di.

```
# To reconstruct this model from state dict, we need to log `a` and `b` along with `MyModel`.
model = MyModel(a=1, b=2)
```

Reply:
@harupy We anticipated the same and that is the reason we have set the argument for `load_state_dict` as  **the model object (not the model class).** Its upto the user to create the model object and call `load_state_dict` method

Reply:
Can we fix `load_state_dict` to just return a state dict because the name `load_state_dict` sounds like it returns a state dict? The workflow would look like:

```python
model = Model(...)
model.load_state_dict(mlflow.pytorch.load_state_dict(...))
```

Reply:
Thanks for the comments. i have updated the code to return state dict.